expression/builtin_time.go

```diff
@@ -850,6 +850,8 @@ func (b *builtinDateFormatSig) evalString(row chunk.Row) (string, bool, error) {
 		return "", true, handleInvalidTimeError(b.ctx, types.ErrWrongValue.GenWithStackByArgs(types.DateTimeStr, t.String()))
 	}
 
+	formatMask = strings.Replace(formatMask, "\n", " ", -1)
+	formatMask = strings.Replace(formatMask, "\r", " ", -1)
```

Comment:
Looks like the root cause is when convert string to date/datetime, `\n` is treated as an invalid character? Because in MySQL 8.0.25
```
mysql> select date_format('2022-02-01', '%Y-%m-%d
16:33:00');
+------------------------------------------------+
| date_format('2022-02-01', '%Y-%m-%d
16:33:00') |
+------------------------------------------------+
| 2022-02-01
16:33:00                            |
+------------------------------------------------+
1 row in set (0.00 sec)

mysql> select cast(date_format('2022-02-01', '%Y-%m-%d
16:33:00') as datetime);
+------------------------------------------------------------------+
| cast(date_format('2022-02-01', '%Y-%m-%d
16:33:00') as datetime) |
+------------------------------------------------------------------+
| 2022-02-01 16:33:00                                              |
+------------------------------------------------------------------+
1 row in set (0.00 sec)
```
In TiDB
```
mysql> select date_format('2022-02-01', '%Y-%m-%d
    '> 16:33:00')16:33:00');
+------------------------------------------------+
| date_format('2022-02-01', '%Y-%m-%d
16:33:00') |
+------------------------------------------------+
| 2022-02-01
16:33:00                            |
+------------------------------------------------+
1 row in set (0.00 sec)

mysql>  select cast(date_format('2022-02-01', '%Y-%m-%d
    '> 16:33:00') as datetime);16:33:00') as datetime);
+------------------------------------------------------------------+
| cast(date_format('2022-02-01', '%Y-%m-%d
16:33:00') as datetime) |
+------------------------------------------------------------------+
| NULL                                                             |
+------------------------------------------------------------------+
1 row in set, 1 warning (0.00 sec)

mysql> show warnings;
+---------+------+-------------------------------------------------+
| Level   | Code | Message                                         |
+---------+------+-------------------------------------------------+
| Warning | 1292 | Incorrect datetime value: '2022-02-01
16:33:00' |
+---------+------+-------------------------------------------------+
1 row in set (0.00 sec)
```